packages/serverless/src/helper/common.ts

```diff
@@ -50,40 +45,38 @@ export function objectToLookupPattern(
 
 export const gql = (...args: any[]): string => args.join('\n');
 
-export const convertToIndexSpecification = <T = TCollectionIndex>(
-  index: TCollectionIndex
-): TCollectionIndexMap<T> => {
-  const result: TCollectionIndexMap<T> = {};
-  for (const key in index) {
-    if (index[key]) {
-      const fieldKey = `document.${key}.value` as keyof TCollectionIndexMap<T>;
-      if (index[key] === ESorting.Asc) {
-        result[fieldKey] =
-          1 as TCollectionIndexMap<T>[keyof TCollectionIndexMap<T>];
-      } else if (index[key] === ESorting.Desc) {
-        result[fieldKey] =
-          -1 as TCollectionIndexMap<T>[keyof TCollectionIndexMap<T>];
-      }
-    }
-  }
-  return result;
-};
+export const convertIndexToMongoFormat = <T = any>(
+  index: Record<string, EIndexType>
+): TCollectionIndexMap<T> =>
+  Object.entries(index).reduce((acc, [key, value]) => {
+    let val = value === EIndexType.Asc ? 1 : -1;
+    return { ...acc, [`document.${key}.value`]: val };
```

Comment:
`document.${key}.value` should be a static method of `ModelDocument` e.g: `ModelDocument::indexKey()`

Reply:
Solved, I add an static method and add on validate for it

```ts
 public static indexKeyFormat(key: string) {
    if (!key) {
      throw new Error('Field name cannot be empty.');
    }
    if (key.startsWith('$')) {
      throw new Error(`Field name '${key}' cannot start with '$'.`);
    }
    if (key.includes('\0')) {
      throw new Error(`Field name '${key}' cannot contain null characters.`);
    }

    return `document.${key}.value`;
  }
```